### PR TITLE
AMD Dockerfile additions

### PR DESCRIPTION
**Pull request by RH-steve-grubb**
This sets the vllm build to a Release build type, builds libsodium,
fixes bad permissions on a logging directory, installs libsodium, and adds a LD_LIBRARY_PATH to fixup python bindings for 2 packages.

### CODE REVIEW

File: Dockerfile.rocm.ubi [hunk 1]
```diff
@@ -178,7 +197,13 @@ RUN --mount=type=bind,from=build_amdsmi,src=/install,target=/install/amdsmi/ \
 RUN umask 002 && \
     useradd --uid 2000 --gid 0 vllm && \
     mkdir -p /licenses && \
-    chmod g+rwx $HOME /usr/src /workspace
+    chmod g+rwx $HOME /usr/src /workspace && \
```
Comment (dtrifiro) [inline Dockerfile.rocm.ubi:200]: Why is this required?

File: Dockerfile.rocm.ubi [hunk 2]
```diff
@@ -178,7 +197,13 @@ RUN --mount=type=bind,from=build_amdsmi,src=/install,target=/install/amdsmi/ \
...
+    chmod 0664 /var/log/rocm_smi_lib
+
+# Install libsodium for Tensorizer encryption
+RUN --mount=type=bind,from=libsodium-builder,src=/usr/src/libsodium,target=/usr/src/libsodium \
+    cd /usr/src/libsodium \
+    && make install
```
Comment (dtrifiro) [inline Dockerfile.rocm.ubi:206]: How about to move this before the `uv pip install` step above?

**Review by dtrifiro — APPROVED** [review]

File: Dockerfile.rocm.ubi (hunk 1 above)
Comment (RH-steve-grubb) [reply Dockerfile.rocm.ubi:200]: Fixes world writable directory permissions in the next line.

File: Dockerfile.rocm.ubi (hunk 2 above)
Comment (RH-steve-grubb) [reply Dockerfile.rocm.ubi:206]: sure

File: Dockerfile.rocm.ubi (hunk 1 above)
Comment (RH-steve-grubb) [reply Dockerfile.rocm.ubi:205]: OK, I think the permissions are right. Marking this resolved.

**Review by dtrifiro — APPROVED** [review]

**Review by NickLucche — APPROVED** [review]
